### PR TITLE
Lobby options for hiding prompts and leaderboards

### PR DESCRIPTION
**Pull request by dqian3**
Add options in lobbies for a "competitive" mode that hides prompt ends and restricts access to the leaderboard if lobby users have not played a prompt.

This implements a simple version: users can still click on the results, but will be redirected back if they haven't played. Also the prompt ends always remain hidden. Later PR will implement a "played" field similar to the main prompts

Closing #455

### CODE REVIEW

File: wikispeedruns/lobbys.py [hunk 1]
```diff
@@ -109,35 +188,56 @@ def add_lobby_prompt(lobby_id: int, start: int, end: int) -> bool:
...
+        LEFT JOIN  (
+            SELECT lobby_id, prompt_id, COUNT(*) AS played
+            FROM lobby_runs
+            WHERE
+                {player[0]}=%(player_value)s
+                AND lobby_id=%(lobby_id)s {"AND prompt_id=%(prompt_id)s" if prompt_id else ""}
+            GROUP BY lobby_id, prompt_id
+        ) r
+        ON r.prompt_id = p.prompt_id AND r.lobby_id = p.lobby_id
+        WHERE p.lobby_id=%(lobby_id)s {"AND p.prompt_id=%(prompt_id)s" if prompt_id else ""}
```
Comment (bricehalder) [inline wikispeedruns/lobbys.py:215]: :suspect:

**Review by bricehalder — COMMENTED** [review]
Looks solid but a screenshot or demo video would be really helpful before approving

**Comment by bricehalder** [comment]
Add option for custom rules/competitive toggle into existing lobbies

File: templates/lobbys/create.html [hunk 2]
```diff
@@ -7,20 +7,47 @@
...
+            <!--
+            <div class="form-check">
+                <input class="form-check-input" type="checkbox" id="req-acc-checkbox" v-model="requireAccount">
+                <label class="form-check-label" for="req-acc-checkbox">
+                    <b>Require accounts</b>
+                    <span class="text-muted fw-light">
+                        disables option to join with passcode, users must have a wikispeedruns account
+                    </span>
+                </label>
+            </div> -->
```
[the remark this answers is not in the document]
Comment (dqian3) [reply templates/lobbys/create.html:45]: I want to add this back in the future, so commented it out for now.

**Pull request merged**